### PR TITLE
Use standard TypeScript configuration

### PR DESCRIPTION
**Pull request by Gudahtt**
The TypeScript configuration has been updated to use the same settings we use in the MetaMask module template. The changes are:

* Enable inline source maps.
* Target ES2017 rather than ES5.
* Allow the use of ES2020 syntax/APIs in the TypeScript source.
* Explicitly declare the root directory, to prevent it from changing accidentally.
* Remove unnecessary type roots and excludes (both included the non-existent `types` directory, and were set to default aside from that).
* Enable `esModuleInterop` for better CommonJS compatibility.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+    "rootDir": "src",
     "sourceMap": true,
     "strict": true,
-    "target": "es5",
-    "typeRoots": ["./types", "./node_modules/@types"],
-    "outDir": "dist"
+    "target": "ES2017"
   },
   "include": ["src/**/*.ts"],
-  "exclude": ["node_modules", "types"]
```
Comment (shanejonas) [inline tsconfig.json:13]: should we keep exclude node_modules? or atleast the module template should probably have that

Comment (Gudahtt) [reply tsconfig.json:13]: It's already excluded by default. [From the docs](https://www.typescriptlang.org/tsconfig#exclude):

>Default: `["node_modules", "bower_components", "jspm_packages"]`, plus the value of `outDir` if one is specified.

File: tsconfig.json [hunk 2]
```diff
@@ -1,14 +1,16 @@
 {
   "compilerOptions": {
-    "moduleResolution": "Node",
     "declaration": true,
+    "esModuleInterop": true,
+    "inlineSources": true,
+    "lib": ["ES2020"],
```
Comment (Gudahtt) [inline tsconfig.json:6]: It looks like this broke the tests. I think I need to include `DOM` here.

Comment (Gudahtt) [reply tsconfig.json]: Addressed in f4b28dcab34066af1d68e389a59eb816582b087d

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**